### DISCUSSION TITLE
Can we Create tunnel without admin rights?

### DISCUSSION THREAD

**Discussion by mega6453** [doronz88/pymobiledevice3#1149] [Q&A] [answered]
@doronz88 
Pymd3 works with iOS 17 by creating tunnel. But the problem is we need admin rights. Is there something you can implement that does not require admin rights? E.g. go-iOS implemented a way to create tunnel without admin rights. So if pymd3 also have this option it will be good. Because every user won't have admin rights. Thank You !

**Comment by doronz88** [comment]
I don't see how this can be achieved without implementing a user-mode IP-stack. If you think otherwise is possible, feel free to submit a PR

**Reply by mega6453** [reply]
Ya may be that's based on user mode IP stack.  But I don't have experience to work on this . Anyway Thanks.

**Reply by mega6453** [reply]
Can you implement to run the tunnel using user-mode IP-stack? 
Use case : In some companies, the windows machine won't have admin permission and they don't provide. So, if it's user-mode IP stack, no need of admin permission. We can run iOS automation easily. So this would be very helpful. Can you please give this solution?

**Reply by doronz88** [reply, marked as answer]
That is very non trivial and complicated to achieve. Also, any bug will be very difficult to debug so I won't implement it myself, but if someone else offer to make a Python package implementing it, I will add an option for it.

BTW, it will also be much much much slower

**Reply by mega6453** [reply]
OK got it, thanks for your reply.

**Comment by nanoscopic** [comment]
As a workaround for this you can setuid a small program that bridges a utun to a uds. I've released the code to do this.

That way, at least, you don't have to run any of the python stuff as root.

go-ios avoids this by using a soft tcp stack instead of a utun, which... is a terrible solution...

on MacOS this is best avoided by reusing the system rsd tunnel instead of trying to make your own
